sherpa/optmethods/tests/test_optmethods.py

```diff
@@ -39,13 +38,22 @@ def tst_all( self, name, fct, fmin, x0, xmin, xmax,
                  iprint=False ):
         self.tst( optfcts.neldermead, name + self.nm, fct, fmin,
                   x0, xmin, xmax, iprint=iprint )
-##         self.tst( stogo.stogo, name + '_stogo', fct, fmin,
-##                   x0, xmin, xmax, iprint=iprint )
         self.tst( optfcts.montecarlo,  name + self.mc , fct, fmin,
                   x0, xmin, xmax, iprint=iprint, maxfev=8192*len(x0) )
         self.tst( optfcts.lmdif, name + self.lm, fct, fmin,
                   x0, xmin, xmax, iprint=iprint )
 
+        #
+        # If the following lines are uncommented out, we get errors
+        # when running the optimization test
```

Comment:
An optimization method, no matter how well written, is not expected to pass
all tests.  There are probably some tweaks that one can do but this was a
low effort on my part and I did not want to investigate it.

On Fri, Mar 25, 2016 at 9:48 AM, Doug Burke <notifications@github.com>
wrote:

> In sherpa/optmethods/tests/test_optmethods.py
> <https://github.com/sherpa/sherpa/pull/194#discussion_r57444875>:
>
> >          self.tst( optfcts.montecarlo,  name + self.mc , fct, fmin,
> >                    x0, xmin, xmax, iprint=iprint, maxfev=8192*len(x0) )
> >          self.tst( optfcts.lmdif, name + self.lm, fct, fmin,
> >                    x0, xmin, xmax, iprint=iprint )
> >
> > +        #
> > +        # If the following lines are uncommented out, we get errors
> > +        # when running the optimization test
>
> Ahhh. That's a more-serious issue than I was hoping for. Do you think it's
> okay for these particular tests to fail with this optimiser, or should they
> pass (or do we need to tweak tolerances)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/sherpa/sherpa/pull/194/files/3d8653dfe45d619b1a8a514d7dda345dd38b655f#r57444875>
>